osu.Game.Rulesets.Osu/Skinning/Legacy/OsuLegacySkinTransformer.cs

```diff
@@ -110,14 +130,18 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
                         };
 
                 case OsuSkinComponents.SpinnerBody:
-                    bool hasBackground = Source.GetTexture("spinner-background") != null;
-
-                    if (Source.GetTexture("spinner-top") != null && !hasBackground)
+                    if (spinnerStyle.Value == SpinnerStyle.NewLegacy)
                         return new LegacyNewStyleSpinner();
-                    else if (hasBackground)
+                    else if (spinnerStyle.Value == SpinnerStyle.OldLegacy)
                         return new LegacyOldStyleSpinner();
 
                     return null;
+
+                case OsuSkinComponents.SpinnerBonusCounter:
+                    if (hasSpinner && hasScoreFont)
+                        return new LegacySpinnerBonusCounter();
```

Comment:
The `hasScoreFont` conditional is required, because `LegacySpinnerBonusCounter` creates a legacy sprite text using `skin.GetDrawableComponet(new HUDSkinComponet(ScoreText))`, which would return null if no font texture exists.

Alternative solutions may be:
 - Handle null legacy sprite texts in `LegacySpinnerBounsCounter`, which feels somewhat odd there in a legacy class.
 - Allowing `LegacySkin` to always create a `LegacySpriteText` even if none exists, which I'm not sure if it's fine either, more odd than the first alternative.

Or by improving usability of `LegacySpriteText` and remove `ScoreText`/`ComboText` skin components, which I had a go on, may feel like too much change over such thing, but I think it's worth it in my opinion: https://github.com/ppy/osu/compare/master...frenzibyte:legacy-sprite-text-fonts
 
 With that on, `LegacySpinnerBonusCounter` can create a legacy sprite text as simple as this:
 ```diff
 -                Child = bonusCounter = ((LegacySpriteText)source.GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.ScoreText))).With(s =>
-                {
-                    s.Alpha = 0f;
-                    s.Anchor = Anchor.TopCentre;
-                    s.Origin = Anchor.Centre;
-                    s.Font = s.Font.With(fixedWidth: false);
-                    s.Scale = new Vector2(SPRITE_SCALE);
-                    s.Y = LegacyCoordinatesContainer.SPINNER_TOP_OFFSET + 299;
-                }),
+                Child = bonusCounter = new LegacySpriteText(source, LegacyFont.Score)
+                {
+                    Alpha = 0f,
+                    Anchor = Anchor.TopCentre,
+                    Origin = Anchor.Centre,
+                    Scale = new Vector2(SPRITE_SCALE),
+                    Y = LegacyCoordinatesContainer.SPINNER_TOP_OFFSET + 299,
+                }.With(s => s.Font.With(fixedWidth: false)),
```